public/app/features/alerting/unified/components/silences/MatchedSilencedRules.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { useDebounce } from 'react-use';
+import { useDispatch } from 'react-redux';
+import { css } from '@emotion/css';
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import { DynamicTable, DynamicTableColumnProps, DynamicTableItemProps } from '../DynamicTable';
+import { RuleState } from '../rules/RuleState';
+import { useCombinedRuleNamespaces } from '../../hooks/useCombinedRuleNamespaces';
+import { Annotation } from '../../utils/constants';
+import { findAlertRulesWithMatchers } from '../../utils/matchers';
+import { fetchAllPromAndRulerRulesAction } from '../../state/actions';
+import { CombinedRule } from 'app/types/unified-alerting';
+import { MatcherFieldValue, SilenceFormFields } from '../../types/silence-form';
+
+type MatchedRulesTableItemProps = DynamicTableItemProps<{
+  matchedRule: CombinedRule;
+}>;
+type MatchedRulesTableColumnProps = DynamicTableColumnProps<{ matchedRule: CombinedRule }>;
+
+export const MatchedSilencedRules = () => {
+  const [matchedAlertRules, setMatchedAlertRules] = useState<MatchedRulesTableItemProps[]>();
+  const formApi = useFormContext<SilenceFormFields>();
+  const dispatch = useDispatch();
+  const { watch } = formApi;
+  const matchers: MatcherFieldValue[] = watch('matchers');
+  const styles = useStyles2(getStyles);
+  const columns = useColumns();
+
+  useEffect(() => {
+    dispatch(fetchAllPromAndRulerRulesAction());
+  }, [dispatch]);
+
+  const combinedNamespaces = useCombinedRuleNamespaces();
+  useDebounce(
+    () => {
+      const matchedRules = combinedNamespaces.flatMap((namespace) => {
+        return namespace.groups.flatMap((group) => {
+          return findAlertRulesWithMatchers(group.rules, matchers);
+        });
+      });
+      setMatchedAlertRules(matchedRules);
+    },
+    500,
+    [combinedNamespaces, matchers]
+  );
+
+  return (
+    <div>
+      <h4>
+        Affected alerts {matchedAlertRules && matchedAlertRules.length > 0 ? `(${matchedAlertRules.length})` : ''}
```

Comment:
I'm not too familiar with the design system; but perhaps a `<Badge>` component could be used here to set the value apart from the header?

<img width="346" alt="image" src="https://user-images.githubusercontent.com/868844/151339858-fb08cca3-024c-4d7a-880f-766cd2f61fec.png">